pkg/utils/tracing/span_test.go

```diff
@@ -1,42 +0,0 @@
-package tracing
```

Comment:
Hard to test it. I hope it's ok that I removed the test? Otherwise would need to build a testing hardness and I'm somewhat short on time.

Reply:
@alexmt This test seems to be a keeper to me.   If so, how about creating an issue and document what needs to be done to restore it?

Reply:
@alexmt @wtam2018 I've refactored the tracing "system" and added a test. PTAL.

Reply:
It looks pretty neat.   

Tracking is turned off by default and it is enabled by setting the env var below.   Should we keep this behavor?

`enabled = os.Getenv("ARGOCD_TRACING_ENABLED") == "1"`

Reply:
Gitops engine is a library so I think ideally it should not depend on any environment variables. It should be up to the integrating application to configure it how it wants to. Argo can easily check the environment variable and inject the logging tracer into the library if tracing is enabled.